test/architectures/test_clip.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+
+import torch
+from test.test_utils import assert_expected, set_rng_seed
+from torchmultimodal.architectures.clip import CLIPArchitecture
+
+
+class TestCLIPArchitecture:
+    @pytest.fixture
+    def start(self):
+        set_rng_seed(1234)
+
+        query_encoder = torch.nn.Linear(5, 3)
+        retrieval_encoder = torch.nn.Linear(4, 3)
```

Comment:
nit: since you are showing off how you made the architecture more general, I would not anchor the names here to a retrieval use case. Instead you could just name them `encoder_a` and `encoder_b` or something, with corresponding keys (just an example, I'm sure you can come up with something better than that 🙂 )

Reply:
Make sense. The relationship "query" and "retrieval" is meaningful for the retrieval task per se. For CLIP model it's just a matter of computing the similarities between the two modalities (or more).